### PR TITLE
Log in button

### PR DESCRIPTION
**Pull request by jessmartin**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -2,77 +2,108 @@ import "./style.css";
...
+// Check if the user is logged in.
+// TODO: Figure out how to do this!
+
+// If user is _not_ logged in, hide the interface.
+//   Currently, it starts out hidden. Nothing to do here.
+
+// If user is logged in, show the log in button.
+//   Currently, it starts out shown. Nothing to do here.
+
+async function main() {
```
Comment (luiscmartinez) [inline src/main.ts:14]: I am thinking we want this main function to return the results of wn.initialise() in order to fix the flickering when  redirecting to fission lobby

File: src/main.ts [hunk 2]
```diff
@@ -2,77 +2,108 @@ import "./style.css";
...
+      console.log("no file system");
+    }
+  });
+}
+
+const logIn = document.querySelector<HTMLInputElement>("#log-in")!;
+logIn.addEventListener("click", async function () {
+  // Initialize WebNative (which will force a log-in)
+  // TODO: Why does this not finish before the items below are hidden?
+  await main();
```
Comment (luiscmartinez) [inline src/main.ts:93]: Use the result of main function
Screen Shot 2022-06-02 at 6 40 05 PM

**Review by luiscmartinez — COMMENTED** [review]
sharing my initial thoughts. Excited to see how you'll handle app state

File: src/main.ts (hunk 1 above)
Comment (jessmartin) [reply src/main.ts:14]: I think you're right!

**Comment by jessmartin** [comment]
After diving into the idea of "Log In" and "Log Out", realized that it was a flawed analogy. In this password-less, key-based auth world, a better analogy, at least for mumblr, is "Connected" or "Disconnected" from a filesystem. I'm going to close this PR as we will _also_ want to re-think the UX in light of this change.

Will open a new PR shortly!

**Comment by luiscmartinez** [comment]
> After diving into the idea of "Log In" and "Log Out", realized that it was a flawed analogy. In this password-less, key-based auth world, a better analogy, at least for mumblr, is "Connected" or "Disconnected" from a filesystem. I'm going to close this PR as we will _also_ want to re-think the UX in light of this change.
> 
> Will open a new PR shortly!

[Coding With Jess EP27: Connection Status
](https://www.youtube.com/watch?v=4s39Moq0TGY)
☝️ Reference video of transitioning from the idea of "Log In" and "Log Out" to "Connected" and "Disconnected" from a filesystem.

**Pull request closed without merging**